### PR TITLE
[Calling][Feature] Bypass Setup Screen Feature

### PR DESCRIPTION
**Pull request by Mohtasim**
## Purpose

* The changes include integration of skip setup screen feature
* Public API changes included
* API for bypass setup screen, camera on by default, mic on by default included
* Error for audio permission included as well

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [x] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [x] UI Changes
  - [ ] Screen captures included
  - [x] Tested for Light/Dark mode
  - [x] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [x] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [x] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open demo app
* Click on settings 
* Look for skip setup screen option
* Look for Call control option

## What to Check
Verify that the following are valid
* if contoso user can join teams & acs call skipping the setup screen

### CODE REVIEW

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/error/ErrorCode.kt [hunk 1]
```diff
@@ -16,6 +16,7 @@ internal class ErrorCode : ExpandableStringEnum<ErrorCode?>() {
         val TURN_MIC_OFF_FAILED = fromString("turnMicOffFailed")
         val NETWORK_NOT_AVAILABLE = fromString("networkNotAvailable")
         val CAMERA_INIT_FAILED = fromString("cameraInitiationFailure")
+        val MIC_PERMISSION_DENIED = fromString("micPermissionDenied")
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/error/ErrorCode.kt:19]: Need to update Mic Permission Error API

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeErrorCode.java [hunk 2]
```diff
@@ -32,6 +32,12 @@ public final class CallCompositeErrorCode extends ExpandableStringEnum<CallCompo
      */
     public static final CallCompositeErrorCode CAMERA_FAILURE = fromString("cameraFailure");
 
+    /**
+     * Dispatched when default microphone control API is accessed without microphone permission being granted.
+     */
+    public static final CallCompositeErrorCode MICROPHONE_PERMISSION_REQUIRED =
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/models/CallCompositeErrorCode.java:38]: Need to update Mic Permission Error API

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/controlbar/ControlBarView.kt [hunk 3]
```diff
@@ -90,6 +95,25 @@ internal class ControlBarView : ConstraintLayout {
...
+                if (it == CallingStatus.NONE || it == CallingStatus.CONNECTING) {
+                    cameraToggle.isEnabled = false
+                    micToggle.isEnabled = false
+                    callAudioDeviceButton.isEnabled = false
+                    moreButton.isEnabled = false
+                    callStatePassedConnecting = false
+                } else {
+                    callStatePassedConnecting = true
+                    updateCamera(viewModel.getCameraStateFlow().value)
+                    // cameraToggle.isEnabled = true
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/controlbar/ControlBarView.kt:110]: Remove this line

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/LocalParticipantViewModel.kt [hunk 4]
```diff
@@ -49,6 +49,7 @@ internal class LocalParticipantViewModel(
         cameraDeviceSelectionStatus: CameraDeviceSelectionStatus,
         camerasCount: Int
     ) {
+
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/calling/localuser/LocalParticipantViewModel.kt:52]: Remove unnecessary change

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/PermissionWarningView.kt [hunk 5]
```diff
@@ -72,7 +72,7 @@ internal class PermissionWarningView : LinearLayout {
             cameraPermissionGranted = false
         } else if (permissionState == PermissionStatus.GRANTED) {
             cameraPermissionGranted = true
-            viewModel.turnCameraOn()
+            // viewModel.turnCameraOn()
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/PermissionWarningView.kt:75]: Remove this line

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/SetupViewModel.kt [hunk 6]
```diff
@@ -34,22 +36,28 @@ internal class SetupViewModel(
...
     fun exitComposite() {
+        // double check here if we need both the action to execute
         dispatchAction(action = CallingAction.CallEndRequested())
         dispatchAction(action = NavigationAction.Exit())
     }
 
     override fun init(coroutineScope: CoroutineScope) {
         val state = store.getCurrentState()
+
+        if (state.localParticipantState.cameraState.operation == CameraOperationalStatus.ON) {
```
Comment (Mohtasim) [inline azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/SetupViewModel.kt:47]: when contoso tries, cameraOnDefault

File: azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/error/ErrorCode.kt (hunk 1 above)
Comment (Mohtasim) [reply azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/error/ErrorCode.kt:19]: Updated

**Review by ShaunaSong — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]
